enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,156 @@
+
+## Motivation
+
+### Goals
+As an OpenStack user, I would like to deploy OpenShift onto pre-existing OpenStack networks, subnets and routing schemes.
+
+### Non-Goals
+There are no changes in expectations regarding publicly addressable parts of the cluster.
+
+## Proposal
+- Installer allow users to provide a subnet that should be used for the cluster. 
+- In order to support provider networks in IPI, the subnet passed to the installer must meet these requirements:
+  - have the capacity and ability for the installer to provision ports on the nodes subnet
+  - have dhcp enabled
+  - the installer must be able to send and recieve https traffic to nodes connected to ports on the nodes subnet
+- In order to support provider networks in UPI, additional documentation will be provided to support this use case
+- For those who do not want the installer to provision their networking resources, we will provide additional UPI documentation.
+- The infrastructure resources owned by the cluster continue to be clearly identifiable and distinct from other resources.
+- Destroying a cluster must make sure that no resources are deleted that didn't belong to the cluster. Leaking resources is preferred over any possibility of deleting non-cluster resources.
+
+
+## User Stories
+1. As an administrator of a private openstack cloud, I use provider networks for my openstack cluster, and would like to be able to install OpenShift in this environment.
+2. As an OpenStack tenant, I want to be able to use pre-configured network resources so that I dont have to create new resources to install an OpenShift cluster.
+
+## Implementation Details Notes/Constraints
+
+### Installer Resources
+If a user wants to use a pre-existing subnet, they can provide the installer with the ID of that subnet. The installer needs to have the authorization and quota to provision the necessary ports and resources on that subnet.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    nodesSubnet: abcd-4321
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+Clusters with provider networks often do not use floating IPs, and may have flat networking schemes. In order to accound for this, we will have to make floating IPs in the cluster optional, as well as the external network. so the first change we will make to address this is to make the parameter `platform.openstack.lbFloatingIP` optional. When unset, the installer will not attempt to attach a floating IP to the API port. Likewise, we will have to make the variable `platform.openstack.externalNetwork` optional. When this is unset, the installer will no longer create and attach a floting IP to the bootstrap node.
```

Comment:
How are we to distinguish between provider networks, where the LB IP and External Network aren't defined, and standard IPI where we actually need those parameters to setup the FIPs.  This seems like another opportunity for operator error, where the User is doing standard IPI and forgets to fill in that informaiton

Reply:
This is really hard to account for, all we can really do is make sure that the UI is simple and the documentation is clear. I dont even think we can validate this pre-flight tbh.